strategies/sendfile.macos.scm

```diff
@@ -7,11 +7,16 @@
 ;; EAGAIN may be signaled even when partial data is sent, but the caller expects EAGAIN
 ;; to mean zero bytes sent, so we return the number of bytes sent when non-zero.
 (define %sendfile-implementation
-  (foreign-lambda* double ((integer src) (integer dst) (double offset) (unsigned-integer to_send))
-    "off_t res = (off_t)to_send;"
-    "if(sendfile(src,dst,(off_t)offset,&res,NULL/*&hdtr*/,0) < 0){"
-    "  if(errno == EAGAIN)"
-    "    C_return(res == 0 ? -2 : (double)(offset + res));"
-    "  else C_return(-1);"
-    "}"
-    "C_return((double)(offset + res));"))
+  (foreign-lambda* ssize_t ((integer src) (integer dst) (unsigned-long offset) (unsigned-long to_send))
```

Comment:
According to [this manpage](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/sendfile.2.html) Mac's `sendfile` accepts `off_t` as the number of bytes to send (and is re-used for the return value) and for the offset. So here we should also use `ssize_t` for both.